### PR TITLE
STONEBLD-1518: Allow disabling plugins from YAML

### PR DESCRIPTION
**Pull request by dwalluck**

### CODE REVIEW

File: java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/build/preprocessor/maven/MavenPrepareCommand.java [hunk 1]
```diff
@@ -87,12 +89,26 @@ private void handleBuild(Path file, boolean topLevel) throws IOException {
         }
     }
 
-    private boolean handlePlugins(List<Plugin> plugins, boolean pluginManagement, boolean topLevel) {
+    private boolean handlePlugins(List<Plugin> plugins, boolean pluginManagement, boolean topLevel) throws IOException {
         boolean modified = false;
+        BuildRecipeInfo info = BuildRecipe.BUILD.getHandler().parse(Path.of(PLUGIN_INFO + "/maven.yaml"));
```
Comment (stuartwdouglas) [inline java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/build/preprocessor/maven/MavenPrepareCommand.java:94]: This is the preprocessor command, you need to use the disabledPlugins command line argument that was passed in, maven.yaml is not present here.

File: pkg/apis/jvmbuildservice/v1alpha1/dependencybuild_types.go [hunk 2]
```diff
@@ -142,6 +142,7 @@ type BuildRecipe struct {
 	AdditionalMemory    int                  `json:"additionalMemory,omitempty"`
 	Repositories        []string             `json:"repositories,omitempty"`
 	AllowedDifferences  []string             `json:"allowedDifferences,omitempty"`
+	DisabledPlugins     string               `json:"disabledPlugins,omitempty"`
```
Comment (stuartwdouglas) [inline pkg/apis/jvmbuildservice/v1alpha1/dependencybuild_types.go:145]: This should really be a string array. I know at the moment it is just used as a comma separated list but it is still better to preserve the structure.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]

File: java-components/build-request-processor/pom.xml [hunk 3]
```diff
@@ -53,6 +53,10 @@
             <groupId>io.quarkiverse.jgit</groupId>
             <artifactId>quarkus-jgit</artifactId>
         </dependency>
+        <dependency>
```
Comment (dwalluck) [inline java-components/build-request-processor/pom.xml:56]: Not sure why this change was required. IDEA seemed to want it.

File: java-components/cache/src/main/java/com/redhat/hacbs/artifactcache/services/RecipeManager.java [hunk 4]
```diff
@@ -131,12 +131,28 @@ public List<BuildToolInfo> getBuildToolInfo(String name) {
...
+
+    public List<String> getDisabledPluginInfo(String name) {
+        List<String> results = new ArrayList<>();
+
+        for (var i : recipeDirs) {
+            var path = i.getDisabledPluginInfo(name);
+
+            if (path.isPresent()) {
+                try {
+                    results.addAll(BuildRecipe.BUILD.getHandler().parse(path.get()).getDisabledPlugins());
```
Comment (dwalluck) [inline java-components/cache/src/main/java/com/redhat/hacbs/artifactcache/services/RecipeManager.java:150]: @stuartwdouglas Is this correct? Merging if multiple recipe dirs...

**Comment by rnc** [comment]
Could we squash to get rid of `Merge remote-tracking branch 'upstream/main' into STONEBLD-1518`  (or use rebase in future?) ?

**Pull request merged**